packages/fxa-settings/src/pages/ConnectAnotherDevice/index.tsx

```diff
@@ -0,0 +1,249 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useEffect } from 'react';
+import { RouteComponentProps, Link, navigate } from '@reach/router';
+import { usePageViewEvent } from '../../lib/metrics';
+import { ENTRYPOINTS } from '../../constants';
+import { HeartsVerifiedImage } from '../../components/images';
+import Banner, { BannerType } from '../../components/Banner';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { REACT_ENTRYPOINT } from '../../constants';
+
+export type ConnectAnotherDeviceProps = {
+  email: string;
+  forceView?: boolean;
+  entrypoint: ENTRYPOINTS; // in Backbone, this came in through the queryparam 'entrypoint' and would be validated in the component, defaulting to FXA Menu entrypoint
+  isSignedIn: boolean; // check if `isSignedInAccount` (on "account"), or get `isVerificationSameBrowser` on the broker
+  showSuccessMessage?: boolean; // check if you have a truthy value for "showSuccessMessage" either on the model, or in a search param by the same name. This is so that if you hit this url, you don't automatically be an inaccurate status message
+  isSignUp: boolean; // we get this by getting the verification reason (or "type") off of the model
+  isSignIn: boolean; // we get this by getting the verification reason (or "type") off of the model
+  canSignIn: boolean; //we check if the user is 1. not signed in, and 2. on a UA that supports sync auth (i.e., Firefox Desktop where version > 40, or Firefox android where version > 43)
+  device?: Devices;
+};
+
+export enum Devices {
+  FIREFOX_MENU = 'Firefox Menu',
+  FIREFOX_ANDROID = 'Firefox Android',
+  FIREFOX_DESKTOP = 'Firefox Desktop',
+  FIREFOX_IOS = 'Firefox Ios',
+  OTHER_ANDROID = 'Other Android',
+  OTHER_IOS = 'Other Ios',
+  OTHER = 'Other',
+}
+export const viewName = 'connect-another-device';
+const ConnectAnotherDevice = ({
+  email,
+  forceView = false,
+  entrypoint,
+  isSignedIn,
+  showSuccessMessage = false,
+  isSignUp,
+  isSignIn,
+  canSignIn,
+  device = Devices.FIREFOX_MENU,
+}: ConnectAnotherDeviceProps & RouteComponentProps) => {
+  usePageViewEvent(viewName, REACT_ENTRYPOINT);
```

Comment:
Should we also include metrics to log how the user is being nudged to connect another device? (as seen in `connect_another_device.js`), or add a note to add these back in later?